### PR TITLE
Spark related changes FBR and PEx

### PR DESCRIPTION
**Pull request by srgg**
The PR Introduces changes required for [Spark Connector](https://github.com/basho/spark-riak-connector) like a Full Bucket Read and Parallel Extract and minor fixes. 

It is reincarnation of https://github.com/basho/riak-java-client/pull/612 and here is a task list from the original PR created after review:
- [x] Merge [merge-dpclient-ts1.3](https://github.com/basho/riak-java-client/tree/merge-dpclient-ts1.3) branch into this one. #621 
- [x] Fix Full Bucket Read Test
- [x] Add Javadocs to for "should provide more details about coverageContext"
- [x] Remove commented out line
- [x] Remove self() function
- [x] Add javadoc for `withReturnBody()` in SecondaryIndexQueryOperation:Builder
- [x] Undo auto bucket cleanup in ITestBase.java, switch tests that need it to extend ITestAutoCleanupBase.java
- [x] Switch bracket style to newlines before and after.

### CODE REVIEW

**Comment by srgg** [comment]
build it please

**Comment by alexmoore** [comment]
Looks pretty good, and alternate coverage tests work (yey).  A few minor notes incoming...

File: src/test/java/com/basho/riak/client/api/commands/indexes/itest/ITestFullBucketRead.java [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+
+                logger.debug("{} keys were returned for {}:\n\t{}",
+                    e.getValue().size(), ce, e.getValue()
+                );
+            }
+        }
+        return results;
+    }
+
+    // TODO: Add test for JSON values
```
Comment (alexmoore) [inline src/test/java/com/basho/riak/client/api/commands/indexes/itest/ITestFullBucketRead.java]: Is this still a todo?

File: src/test/java/com/basho/riak/client/api/commands/indexes/itest/ITestFullBucketRead.java [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+                builder.append(String.format("\n\t%s:%d ('%s')",
+                    ce.getHost(),
+                    ce.getPort(),
+                    ce.getDescription()
+                ));
+            }
+            logger.info(builder.toString());
+        }
+
+        // To be sure that
```
Comment (alexmoore) [inline src/test/java/com/basho/riak/client/api/commands/indexes/itest/ITestFullBucketRead.java]: To be sure that what?  The suspense is killing me :)

File: src/main/java/com/basho/riak/client/api/commands/kv/CoveragePlan.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+package com.basho.riak.client.api.commands.kv;
+
+import com.basho.riak.client.api.RiakCommand;
+import com.basho.riak.client.api.commands.CoreFutureAdapter;
+import com.basho.riak.client.core.RiakCluster;
+import com.basho.riak.client.core.RiakFuture;
+import com.basho.riak.client.core.operations.CoveragePlanOperation;
+import com.basho.riak.client.core.query.Namespace;
+
+// TODO: documentation for CoveragePlan needs to be improved
```
Comment (alexmoore) [inline src/main/java/com/basho/riak/client/api/commands/kv/CoveragePlan.java]: Do the Javadocs need improving, or docs on the docs.basho.com site?

We should also make a note somewhere that the Coverage/PEX is only available in TS 1.3/1.4 right now, and not KV.

File: src/main/java/com/basho/riak/client/core/operations/CoveragePlanOperation.java [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+import com.basho.riak.protobuf.RiakKvPB;
+import com.basho.riak.protobuf.RiakMessageCodes;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.util.*;
+
+// TODO: documentation for CoveragePlanOperation needs to be improved
```
Comment (alexmoore) [inline src/main/java/com/basho/riak/client/core/operations/CoveragePlanOperation.java]: See previous note about Coverage docs.

**Comment by alexmoore** [comment]
+1 with the requested docs changes and fix/remove TODOs.

File: src/main/java/com/basho/riak/client/api/commands/kv/CoveragePlan.java (hunk 3 above)
Comment (mdigan) [inline src/main/java/com/basho/riak/client/api/commands/kv/CoveragePlan.java]: @ph07 @laurenrother as @alexmoore mentions, note that when we add Parallel Extract support to the Riak Java Client we'll need to update docs to say the feature is only supported in TS.

**Comment by srgg** [comment]
build it please

**Comment by alexmoore** [comment]
Buildbot isn't picking up the last change, but 05fbe31 [was successful](http://buildbot.bos1/builders/java-client-test/builds/262), only changes since then are comment related.  

+1 merge it.

**Pull request merged**